lib/gui/os/window-progress.js

```diff
@@ -38,12 +38,15 @@ exports.currentWindow = electron.remote.getCurrentWindow();
  * Show progress inline in operating system task bar
  *
  * @param {Number} percentage - percentage
+ * @param {String} status - flashing/validating status
  *
  * @example
  * windowProgress.set(85);
  */
-exports.set = (percentage) => {
+exports.set = (percentage, status) => {
   exports.currentWindow.setProgressBar(utils.percentageToFloat(percentage));
+  const statusString = status === 'check' ? 'Validating' : 'Flashing';
+  exports.currentWindow.setTitle(`Etcher – ${statusString} ${percentage}%`);
```

Comment:
Rather than hardcoding `Etcher` here (and below), we should probably get this from `displayName` in `package.json` ?